src/main/java/seedu/address/logic/commands/StatusCommand.java

```diff
@@ -88,4 +72,24 @@ public boolean equals(Object other) {
         return index.equals(e.index)
                 && status.equals(e.status);
     }
+
+    @Override
+    protected CommandResult executeUndoableCommand(Model model,
+                                                   CommandHistory commandHistory,
+                                                   StackUndoRedo undoRedoStack) throws CommandException {
```

Comment:
None of `StackUndoRedo` is being used in any of the `UndoableCommand`

Reply:
its being used in the execute command function

Reply:
> its being used in the execute command function

That's my point -- so why do you need to change the method signature of all the rest of the commands? This is totally unnecessary and introducing additional complexity to the code, and for no good reason.

Read:
> Another point is that, why do you need to change CommandResult#execute to CommandResult#executeUndoable? You do not make use of the StackUndoRedo in any of these, so method signature should not change. The only places  where you seem to have used UndoRedoStack is in UndoCommand and RedoCommand, which I also doubt needs to  have their method signatures changed. You could just maintain a static reference to your undo and redo stacks.